testing/test_warnings.py

```diff
@@ -591,3 +591,62 @@ def test_custom_warning_formatter():
     )
     result = testdir.runpytest_subprocess()
     result.stdout.fnmatch_lines(["*1 passed, * warnings in*"])
+
+
+class TestAssertionWarnings:
+    @staticmethod
+    def assert_result_warns(result):
+        result.stdout.fnmatch_lines(["*PytestWarning*"])
+
+    def test_tuple_warning(self, testdir):
+        testdir.makepyfile(
+            """
+            def test_foo():
+                assert (1,2)
+            """
+        )
+        result = testdir.runpytest()
+        self.assert_result_warns(result)
+
+    @staticmethod
+    def create_file(testdir, return_none):
+        testdir.makepyfile(
+            """
+            def foo(return_none):
+                if return_none:
+                    return None
+                else:
+                    return False
+
+            def test_foo():
+                assert foo({return_none})
+            """.format(
+                return_none=return_none
+            )
+        )
+
+    def test_none_function_warns(self, testdir):
+        self.create_file(testdir, True)
+        result = testdir.runpytest()
+        self.assert_result_warns(result)
+
+    @pytest.mark.xfail(strict=True)
```

Comment:
Instead of using `xfail`, please use a proper assertion that we are not receiving any warnings, something like:

```python
result.stdout.fnmatch_lines(["*1 passed in*"])
```

Because warnings will generate a different summary ("1 passed, 1 warnings in"), which would fail the test. Same for `test_false_function_no_warn` below.

Reply:
Right! I saw that later with other tests in the same module, I'll get to it